api-reference/10 UI Widgets/GridBase/1 Configuration/rowDragging/allowReordering.md

```diff
@@ -8,7 +8,14 @@ default: false
 Allows row reordering using drag and drop gestures.
 
 ---
-This option only enables dragging. You should also implement the reordering logic in the [onReorder]({basewidgetpath}/Configuration/rowDragging/#onReorder) handler to enable row reordering inside a component. If you need to perform cross-component drag and drop, implement the dragging logic in the [onAdd]({basewidgetpath}/Configuration/rowDragging/#onAdd) handler.
+This option requires that also you implement the [onReorder]({basewidgetpath}/Configuration/rowDragging/#onReorder) handler to allow users to reorder rows inside a single component. 
+
+To perform cross-component drag and drop, implement at least one of the following handlers instead of the **onReorder**:     
+
+- [onRemove]({basewidgetpath}/Configuration/rowDragging/#onRemove) to handle start of item dragging from one component 
+- [onAdd]({basewidgetpath}/Configuration/rowDragging/#onAdd) to handle item drop to another component
+
+If you need to forbid users to drag items inside or between the components, specify the rules in the [onDragChange]({basewidgetpath}/Configuration/rowDragging/#onDragChange) handler. 
```

Comment:
I went to the onDragChange description and didn't found any further information on the use case. Furthermore, I believe that to forbid users to drg items, I can simply keep allowReordering disabled, why bother with onDragChange?

I also had a look at the [Drag & Drop Between Grids](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/DnDBetweenGrids/Angular/Light/) demo, and it looks like allowReordering has nothing to do with cross-component drag and drop. Why provide so many details on this use case here?

Reply:
1. My description was not certain enough. It should be "If you need to forbid users to drag items inside or between the **certain** components, specify the rules..." in the same way as it is described in the  [Node d&d demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeList/LocalReordering/jQuery/Light/). Also I think that I should add a snippet on the onDragChange description.

2. Previously, I just haven't found a better topic to place this info and followed the task as it was written. Alyar suggested moving this to the [group](https://js.devexpress.com/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/rowDragging/#group) option. What do you think?